ci/qemu_test.py

```diff
@@ -89,3 +99,42 @@ def test_password_reset_required(vm):
     vm.spawn.expect_exact("Retype new password:")
     vm.spawn.send("new password\r\n")
     vm.spawn.expect_exact("debian@debian:~$")
+
+    vm.logged_in = True
+
+
+@pytest.fixture(scope="module")
+def logged_in_vm(vm):
+    if not vm.logged_in:
+        pytest.skip("Password reset test did not run or failed")
+    return vm
+
+
+def test_using_guest_tests(logged_in_vm):
+    """Run the tests in qemu_guest_test.py inside the qemu guest"""
+    # Statement of test success and failure that are unlikely to appear by
+    # accident
+    SUCCESS_NOTICE = "All ci/qemu_guest_test.py tests passed"
+    FAILURE_NOTICE = "Some ci/qemu_guest_test.py tests failed"
+    # We use apt-get -U here and the apt_dependencies fixture in
+    # qemu_guest_test.py relies on this.
+    SCRIPT = f"""sudo -i sh <<EOT
+apt-get install -Uy --no-install-recommends python3-pytest
```

Comment:
It's quite unfortunate to introduce a network dependency :-(

I also don't like that we're changing the rootfs, but I guess we need to bring the tests somehow.

Should we just add this test runner to the image?